synapse/handlers/federation.py

```diff
@@ -156,7 +156,7 @@ def __init__(self, hs):
             self._maybe_store_room_on_invite = self.store.maybe_store_room_on_invite
 
         # When joining a room we need to queue any events for that room up
-        self.room_queues = {}
+        self.room_queues = {}  # type: Dict[str, List[Tuple[str, EventBase]]]
```

Comment:
I think this is a `Dict[str, List[Tuple[EventBase, str]]]`, this seems to match where the queue gets processed:

https://github.com/matrix-org/synapse/blob/a35bf130381833a66bf8dce4cd4078259914528c/synapse/handlers/federation.py#L1439-L1440

Reply:
so it is. Why doesn't mypy pick this up...